src/main/java/io/boomerang/service/refactor/TaskServiceImpl.java

```diff
@@ -544,7 +523,7 @@ private void updatePendingApprovalStatus(ActivityEntity workflowActivity) {
         ApprovalStatus.submitted);
     boolean existingApprovals = (count > 0);
     workflowActivity.setAwaitingApproval(existingApprovals);
-    activityService.saveWorkflowActivity(workflowActivity);
+    workflowActivity = activityService.saveWorkflowActivity(workflowActivity);
```

Comment:
Re assign this, as the .saveWorkflowActivity(..) may return a different instance of this entity.